lib/std/Build/Watch.zig

```diff
@@ -237,6 +237,230 @@ const Os = switch (builtin.os.tag) {
             }
         }
     },
+    .windows => struct {
+        const posix = std.posix;
+        const windows = std.os.windows;
+
+        /// Keyed differently but indexes correspond 1:1 with `dir_table`.
+        handle_table: HandleTable,
+        handle_extra: std.MultiArrayList(HandleExtra),
+
+        const HandleTable = std.AutoArrayHashMapUnmanaged(windows.HANDLE, ReactionSet);
+        const HandleExtra = struct {
+            dir: *Directory,
+            wait_handle: windows.HANDLE,
+        };
+        const Directory = struct {
+            handle: windows.HANDLE,
+            overlapped: windows.OVERLAPPED,
+            buffer: [64512]u8 align(@alignOf(windows.FILE_NOTIFY_INFORMATION)) = undefined,
+
+            fn readChanges(self: *@This()) void {
+                const notify_filter =
+                    windows.FILE_NOTIFY_CHANGE_CREATION |
+                    windows.FILE_NOTIFY_CHANGE_DIR_NAME |
+                    windows.FILE_NOTIFY_CHANGE_FILE_NAME |
+                    windows.FILE_NOTIFY_CHANGE_LAST_WRITE |
+                    windows.FILE_NOTIFY_CHANGE_SIZE;
+                const r = windows.kernel32.ReadDirectoryChangesW(self.handle, @ptrCast(&self.buffer), self.buffer.len, 0, notify_filter, null, &self.overlapped, null);
+                assert(r != 0);
```

Comment:
This assert doesn't seem like a good idea, since `ReadDirectoryChangesW` can presumably fail legitimately. For example, [from the docs](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-readdirectorychangesw):

```
If the network redirector or the target file system does not support this operation, the function fails with ERROR_INVALID_FUNCTION.
```

Instead of an assertion, it would be better to make `readChanges` able to return an error and handle the `ReadDirectoryChangesW` return like so:

```zig
    if (r == 0) {
        switch (windows.GetLastError()) {
            .INVALID_FUNCTION => error.ReadDirectoryChangesUnsupported,
            else => |err| return unexpectedError(err),
        }
    }
```